src/tls.c

```diff
@@ -4479,99 +4479,176 @@ static byte* TLSX_QSHKeyFind_Pub(QSHKey* qsh, word16* pubLen, word16 name)
+                    &public_key_len, WOLFSSL_NTRU_EESS439);
+            }
+            if (TLSX_UseQSHScheme(&ssl->extensions, WOLFSSL_NTRU_EESS439,
+                                  public_key, public_key_len, ssl->heap)
+                                  != SSL_SUCCESS)
+                ret = -1;
+        }
+        else if (ssl->sendQSHKeys && ssl->QSH_Key == NULL) {
+            /* for each scheme make a client key */
+            extension = TLSX_Find(ssl->extensions, TLSX_QUANTUM_SAFE_HYBRID);
+            if (extension) {
+                qsh = (QSHScheme*)extension->data;
+
+                while (qsh) {
+                    if ((ret = TLSX_CreateQSHKey(ssl, qsh->name)) != 0)
+                        return ret;
 
-                /* add NTRU 128 */
-                if (ssl->sendQSHKeys) {
+                    /* get next now because qsh could be freed */
+                    next = qsh->next;
+
+                    /* find the public key created and add to extension*/
                     public_key = TLSX_QSHKeyFind_Pub(ssl->QSH_Key,
-                        &public_key_len, WOLFSSL_NTRU_EESS439);
-                }
-                if (TLSX_UseQSHScheme(&ssl->extensions, WOLFSSL_NTRU_EESS439,
-                                      public_key, public_key_len, ssl->heap)
-                                      != SSL_SUCCESS)
-                    ret = -1;
-            }
-            else if (ssl->sendQSHKeys && ssl->QSH_Key == NULL) {
-                /* for each scheme make a client key */
-                extension = TLSX_Find(ssl->extensions, TLSX_QUANTUM_SAFE_HYBRID);
-                if (extension) {
-                    qsh = (QSHScheme*)extension->data;
-
-                    while (qsh) {
-                        if ((ret = TLSX_CreateQSHKey(ssl, qsh->name)) != 0)
-                            return ret;
-
-                        /* get next now because qsh could be freed */
-                        next = qsh->next;
-
-                        /* find the public key created and add to extension*/
-                        public_key = TLSX_QSHKeyFind_Pub(ssl->QSH_Key,
-                                 &public_key_len, qsh->name);
-                        if (TLSX_UseQSHScheme(&ssl->extensions, qsh->name,
-                                              public_key, public_key_len,
-                                              ssl->heap) != SSL_SUCCESS)
-                            ret = -1;
-                        qsh = next;
-                    }
+                             &public_key_len, qsh->name);
+                    if (TLSX_UseQSHScheme(&ssl->extensions, qsh->name,
+                                          public_key, public_key_len,
+                                          ssl->heap) != SSL_SUCCESS)
+                        ret = -1;
+                    qsh = next;
                 }
             }
-         } /* is not server */
-    #endif
+        }
+#else
```

Comment:
I don't think that this should be an else case. QSH extension is 0x17 and elliptic curves extension is 0x0a. Both can be sent with the same Client Hello.